### PR TITLE
enhance: emit an error when the Auth method is specified but no creds were passed

### PR DESCRIPTION
**Pull request by colindean**
This avoids a frustrating timeout error when the secrets weren't passed correctly.

### CODE REVIEW

File: cmd/vela-email/plugin_test.go [hunk 1]
```diff
@@ -279,6 +301,20 @@ func TestValidateErrors(t *testing.T) {
 			},
 			wantErr: ErrorMissingSMTPParam,
 		},
+		{
+			name: "SMTP auth supplied but credentials missing",
+			parameters: Plugin{
+				Auth: "LoginAuth",
+				SMTPHost: &SMTPHost{
+					Host: "smtphost.com",
+					Port: "234234",
```
Comment (colindean) [inline cmd/vela-email/plugin_test.go:310]: Also LOL this works with an invalid port, that may be a problem if something else doesn't check that

**Review by wass3r — DISMISSED** [review]
lgtm - thanks!

File: cmd/vela-email/plugin.go [hunk 2]
```diff
@@ -31,6 +31,9 @@ var (
 
 	// ErrorMissingSMTPParam is returned when the plugin is missing a smtp host or port parameter.
 	ErrorMissingSMTPParam = errors.New("missing smtp parameter (host/port)")
+
+	// ErrorAuthSpecifiedButCredentialsMissing is returned when the plugin is missing credentials when auth type was specified
```
Comment (colindean) [inline cmd/vela-email/plugin.go:35]:
```suggestion
	// ErrorAuthSpecifiedButCredentialsMissing is returned when the plugin is missing credentials when auth type was specified
```
```suggestion
	// ErrorAuthSpecifiedButCredentialsMissing is returned when the plugin is missing credentials when auth type was specified.
```

File: cmd/vela-email/plugin.go [hunk 3]
```diff
@@ -141,6 +144,10 @@ func (p *Plugin) Validate() error {
 		return ErrorMissingSMTPParam
 	}
 
+	if len(p.Auth) > 0 && !(len(p.SMTPHost.Username) > 0 && len(p.SMTPHost.Username) > 0) {
```
Comment (colindean) [inline cmd/vela-email/plugin.go:147]:
```suggestion
	if len(p.Auth) > 0 && !(len(p.SMTPHost.Username) > 0 && len(p.SMTPHost.Password) > 0) {
```

Comment (wass3r) [reply cmd/vela-email/plugin.go]: yikes, sorry i missed that :shame_on_me:

**Review by wass3r — APPROVED** [review]

**Comment by colindean** [comment]
Looks like the check failure is just a copyright date. Perhaps that can be done in another PR.

**Comment by colindean** [comment]
@wass3r Who else can I or you tag to review?

**Comment by wass3r** [comment]
@go-vela/admins should do. I'll bug them tomorrow ;)

**Review by ecrupper — APPROVED** [review]

**Pull request merged**